sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt

```diff
@@ -396,4 +397,42 @@ internal class KlaviyoLocationManager(
             createEvent(event, getAsProfile())
         }.uniqueId
     }
+
+    /**
+     * Handle device boot event to re-register geofences
+     *
+     * System geofences are cleared on device reboot, so we need to restore them from
+     * persistent storage. This method checks for location permissions and re-registers
+     * any stored geofences if permissions are granted.
+     *
+     * Note: This requires Klaviyo to have been initialized at least once before the device reboot
+     * so that Config, dataStore, and location permissions are accessible. If Klaviyo has never been
+     * initialized, this method will silently return without restoring geofences.
+     */
+    @SuppressLint("MissingPermission")
+    override fun restoreGeofencesOnBoot(context: Context) {
+        // Check if we have location permissions
+        if (!KlaviyoPermissionMonitor.hasGeofencePermissions(context)) {
+            Registry.log.info("Location permissions not granted, skipping geofence restoration")
+            return
+        }
+
+        Registry.log.info("Handling device boot event to restore geofences")
+
+        if (!Registry.isRegistered<Config>()) {
+            // Initialize Klaviyo with config to access dataStore
+            Klaviyo.registerForLifecycleCallbacks(context)
+        }
+
+        // Re-register geofences with the system
+        getStoredGeofences()
```

Comment:
Should we pull from the API instead incase something has changed? I guess when app is launched this happens anyways?

Reply:
Actually, I don't think I can do that. In this boot receiver, they either won't have provided a company ID yet, or else they'll have called `registerGeofencing` in their application launch code which will do the same thing anyway.